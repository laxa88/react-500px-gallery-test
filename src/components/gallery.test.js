import React from 'react';
import 'jest';
import 'enzyme';

import Gallery from './gallery';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Gallery', () => {
  test('Can render', () => {
    const wrapper = shallow(
      <Gallery
        photos={[
          {id: 1, image_url: 'dummyUrl1', name: 'item 1'},
          {id: 2, image_url: 'dummyUrl2', name: 'item 2'},
          {id: 3, image_url: 'dummyUrl3', name: 'item 3'},
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Shows message when no photos are provided', () => {
    const wrapper = shallow(
      <Gallery
        photos={null}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
