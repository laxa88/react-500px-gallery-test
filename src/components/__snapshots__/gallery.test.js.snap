// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gallery
    photos={
      Array [
        Object {
          "id": 1,
          "image_url": "dummyUrl1",
          "name": "item 1",
        },
        Object {
          "id": 2,
          "image_url": "dummyUrl2",
          "name": "item 2",
        },
        Object {
          "id": 3,
          "image_url": "dummyUrl3",
          "name": "item 3",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="cell"
        >
          <div>
            <img
              src="dummyUrl1"
              title="item 1"
            />
          </div>
        </div>,
        <div
          className="cell"
        >
          <div>
            <img
              src="dummyUrl2"
              title="item 2"
            />
          </div>
        </div>,
        <div
          className="cell"
        >
          <div>
            <img
              src="dummyUrl3"
              title="item 3"
            />
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <img
              src="dummyUrl1"
              title="item 1"
            />
          </div>,
          "className": "cell",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="dummyUrl1"
              title="item 1"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "dummyUrl1",
              "title": "item 1",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <img
              src="dummyUrl2"
              title="item 2"
            />
          </div>,
          "className": "cell",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="dummyUrl2"
              title="item 2"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "dummyUrl2",
              "title": "item 2",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <img
              src="dummyUrl3"
              title="item 3"
            />
          </div>,
          "className": "cell",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="dummyUrl3"
              title="item 3"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "dummyUrl3",
              "title": "item 3",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cell"
          >
            <div>
              <img
                src="dummyUrl1"
                title="item 1"
              />
            </div>
          </div>,
          <div
            className="cell"
          >
            <div>
              <img
                src="dummyUrl2"
                title="item 2"
              />
            </div>
          </div>,
          <div
            className="cell"
          >
            <div>
              <img
                src="dummyUrl3"
                title="item 3"
              />
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <img
                src="dummyUrl1"
                title="item 1"
              />
            </div>,
            "className": "cell",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                src="dummyUrl1"
                title="item 1"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "dummyUrl1",
                "title": "item 1",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <img
                src="dummyUrl2"
                title="item 2"
              />
            </div>,
            "className": "cell",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                src="dummyUrl2"
                title="item 2"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "dummyUrl2",
                "title": "item 2",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <img
                src="dummyUrl3"
                title="item 3"
              />
            </div>,
            "className": "cell",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                src="dummyUrl3"
                title="item 3"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "dummyUrl3",
                "title": "item 3",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Gallery Shows message when no photos are provided 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gallery
    photos={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        No results found.
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No results found.",
      },
      "ref": null,
      "rendered": "No results found.",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          No results found.
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No results found.",
        },
        "ref": null,
        "rendered": "No results found.",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
